modules/core/src/passes/layers-pass.ts

```diff
@@ -226,18 +241,42 @@ export default class LayersPass extends Pass {
       viewport
     });
 
-    if (view && view.props.clear) {
-      const clearOpts = view.props.clear === true ? {color: true, depth: true} : view.props.clear;
-      const clearRenderPass = this.device.beginRenderPass({
-        framebuffer: target,
-        parameters: {
-          viewport: glViewport,
-          scissorRect: glViewport
-        },
-        clearColor: clearOpts.color ? [0, 0, 0, 0] : false,
-        clearDepth: clearOpts.depth ? 1 : false
-      });
-      clearRenderPass.end();
+    if (view) {
+      const {clear, clearColor, clearDepth, clearStencil} = view.props;
+      if (clear) {
+        // If clear option is set, clear all buffers by default.
+        let colorToUse: NumberArray4 | false = [0, 0, 0, 0];
+        let depthToUse: number | false = 1.0;
+        let stencilToUse: number | false = 0;
+
+        if (Array.isArray(clearColor)) {
+          colorToUse = [...clearColor.slice(0, 3), clearColor[3] || 255].map(
+            c => c / 255
+          ) as NumberArray4;
+        } else if (clearColor === false) {
+          colorToUse = false;
+        }
+
+        if (clearDepth !== undefined) {
+          depthToUse = clearDepth;
+        }
+
+        if (clearStencil !== undefined) {
+          stencilToUse = clearStencil;
+        }
+
+        const clearRenderPass = this.device.beginRenderPass({
```

Comment:
We probably do not care about WebGPU right now, but I don't think this is going to work on WebGPU. 

- The commands recorded in the render passes are going to be injected into the command stream at the location of beginRenderPass.
- So, we need to begin this clear render pass before the render pass that has been passed in.
- If we can do that, we can probably just apply the clear parameters to the creation of that render pass.